packages/components/src/components/vertical-tab/vertical-tab-headers.tsx

```diff
@@ -1,21 +1,43 @@
-import classNames from 'classnames';
 import React from 'react';
-import VerticalTabWrapper from './vertical-tab-wrapper.jsx';
-import VerticalTabHeader from './vertical-tab-header.jsx';
-import VerticalTabHeaderGroup from './vertical-tab-header-group.jsx';
-import VerticalTabHeaderTitle from './vertical-tab-header-title.jsx';
+import classNames from 'classnames';
+import VerticalTabWrapper from './vertical-tab-wrapper';
+import VerticalTabHeader, { TItem } from './vertical-tab-header';
+import VerticalTabHeaderGroup from './vertical-tab-header-group';
+import VerticalTabHeaderTitle from './vertical-tab-header-title';
+
+type TVerticalTabHeaders = {
+    className?: string;
+    extra_offset?: number;
+    has_mixed_dimensions?: boolean;
+    header_title?: string;
+    is_collapsible?: boolean;
+    is_floating?: boolean;
+    is_routed?: boolean;
+    item_groups?: TItem[];
+    items: TItem[];
+    onChange: (item: TItem) => void;
+    selected: TItem;
+    selectedKey?: string;
+};
 
-const offsetTop = (extra_offset, is_floating, ref, selected) => {
+const offsetTop = (
+    extra_offset: number | undefined,
+    is_floating: boolean | undefined,
```

Comment:
```suggestion
    extra_offset?: number,
    is_floating?: boolean,
```

Reply:
Particularly in this case it is not working, because for this function 'extra_offset' and 'is_floating' shouldn’t  be optional params. I got this error:
<img width="549" alt="Screenshot 2023-02-20 at 11 04 05" src="https://user-images.githubusercontent.com/103182683/220047362-043e5335-d521-4267-b8a5-d1bfbdc3b572.png">
you can see here the explanation:
https://stackoverflow.com/questions/46958782/why-does-typescript-require-optional-parameters-after-required-parameters